src/main/java/com/cst438/service/GradebookServiceProxy.java

```diff
@@ -0,0 +1,254 @@
+package com.cst438.service;
+
+import com.cst438.dto.*;
+import com.cst438.domain.*;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+@Service
+public class GradebookServiceProxy {
+
+    @Autowired
+    private CourseRepository courseRepository;
+
+    @Autowired
+    private SectionRepository sectionRepository;
+
+    @Autowired
+    private EnrollmentRepository enrollmentRepository;
+
+    @Autowired
+    private TermRepository termRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    Queue gradebookServiceQueue = new Queue("gradebook_service", true);
+
+    @Bean
+    public Queue createQueue() {
+        return new Queue("registrar_service", true);
+    }
+
+    @Autowired
+    RabbitTemplate rabbitTemplate;
+
+    @RabbitListener(queues = "registrar_service")
+    public void receiveFromRegistrar(String message) {
+        try {
+            MessageWrapper wrapper = fromJsonString(message, MessageWrapper.class);
```

Comment:
I don't think this is going to work. He has an example of a much simpler way of doing this.

Reply:
You can see how it's done on my branch